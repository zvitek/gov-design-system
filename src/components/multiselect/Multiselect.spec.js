import { shallowMount } from '@vue/test-utils'
import GovMultiselect from '@components/Multiselect/Multiselect'
import GovField from '@components/Field/Field'
import GovAutocomplete from '@components/Autocomplete/Autocomplete'

let wrapper

describe('GovMultiselect', () => {
    beforeEach(() => {
        wrapper = shallowMount(GovMultiselect, {
            propsData: {
                data: []
            },
            components: {
                GovField, GovAutocomplete
            }
        })
    })

    it('is called', () => {
        expect(wrapper.name()).toBe('GovMultiselect')
        expect(wrapper.isVueInstance()).toBeTruthy()
    })

    it('render correctly', () => {
        const wrapper = shallowMount(GovMultiselect, {
            propsData: {
                data: [],
                customUniqueId: 'Multiselect_abc'
            }
        })
        expect(wrapper.html()).toMatchSnapshot()
    })

    it('has a input element', () => {
        expect(wrapper.find(GovAutocomplete)).toBeTruthy()
    })
})
