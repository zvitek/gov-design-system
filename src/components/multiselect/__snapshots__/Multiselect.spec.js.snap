// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GovMultiselect render correctly 1`] = `
<gov-field>
    <gov-autocomplete data=""></gov-autocomplete>
    <ul class="gov-multiselect__tags"></ul>
</gov-field>
`;
